### ISSUE TITLE
It's not really clear you need a semaphore per frame

### ISSUE THREAD

**Issue by darksylinc** [LunarG/VulkanSamples#268]
Hi!

Back when I was studying VulkanSamples, I noticed the samples always do this:

```
res = vkAcquireNextImageKHR(info.device, info.swap_chain, UINT64_MAX, imageAcquiredSemaphore, VK_NULL_HANDLE,
                            &info.current_buffer);
```

Analyzing the code it would look that only a semaphore is needed. However this is not true.
VulkanSamples get away with this because almost all samples either render a frame or wait for every frame to complete before submitting the next one.

I get that stalling per frame is done for simplicity and I applaud that, however this is can cause confusion which in turn leads to bugs when copying the samples' code.

For the basic samples this could be easily fixed with a comment:

```
// Note in a real application you will need one semaphore per frame! e.g. imageAcquiredSemaphore[frameIdx] !
res = vkAcquireNextImageKHR(info.device, info.swap_chain, UINT64_MAX, imageAcquiredSemaphore, VK_NULL_HANDLE,
                            &info.current_buffer);
```

When we take a deeper look at Hologram's sample, we also see it performs:

```
res = vk::AcquireNextImageKHR(ctx_.dev, ctx_.swapchain, UINT64_MAX, buf.acquire_semaphore, VK_NULL_HANDLE, &buf.image_index);
```

This is sort of correct because `buf` comes from ctx_.back_buffers (an array). However this is not as obvious as acquire_semaphore[frameIdx].
I have to say I missed this several times and thought LunarG's samples were wrong because of not being obvious enough until yesterday.

I said 'sort of correct' because the Hologram sample ties buffering with the number of backbuffers. Thus a GPU which returns a minimum of 4 back buffers will end up performing quadruple buffer; when a simple double buffer scheme could suffice.

The number of semaphores required in a double buffer scheme is 2 (as long as the fences work correctly to prevent the CPU from going too far ahead), irrespective of the minimum number of backbuffers the present queue returns.

I don't know what to say about it. Vulkan is often targeted to 'experts' thus any expert should be able to notice this and already know the difference between buffering frames and the minimum number of backbuffers. 

But on the other hand many beginners go straight into Vulkan and copy the samples. And if the samples aren't very clear about it, bad usage gets spread out.

TL;DR: This ticket is not about the samples being wrong, but rather about being misleading or giving the wrong impression

**Comment by Tony-LunarG** [comment]
Hologram will try to use however many back buffers are specified in settings, with the default being 1 back buffer (and it will add another buffer to render into).  So by default Hologram will be double buffering, independent of how many swapchain images are minimally required.  I acknowledge that Hologram is not the easiest code to read.
